### PR TITLE
[wip] Azure support

### PR DESCRIPTION
**Pull request by sylfabre**

### CODE REVIEW

**Comment by sylfabre** [comment]
@makasim my colleague @PaulRouss3l is working on it now. He'll do it tomorrow.

File: pkg/azure-storage/AzureStorageConsumer.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function reject(Message $message, bool $requeue = false): void
+    {
+        InvalidMessageException::assertMessageInstanceOf($message, AzureStorageMessage::class);
+
+        if(false === $requeue) {
```
Comment (makasim) [inline pkg/azure-storage/AzureStorageConsumer.php]: it should publish a message on requeue == true

File: pkg/azure-storage/AzureStorageProducer.php [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            'nextTimeVisible' => $resultMessage->getTimeNextVisible(),
+        ]);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setDeliveryDelay(int $deliveryDelay = null): Producer
+    {
+        throw new DeliveryDelayNotSupportedException();
```
Comment (makasim) [inline pkg/azure-storage/AzureStorageProducer.php]: Do not throw exception if null is passed

File: pkg/azure-storage/AzureStorageProducer.php [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    {
+        return null;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setPriority(int $priority = null): Producer
+    {
+        throw new PriorityNotSupportedException();
```
Comment (makasim) [inline pkg/azure-storage/AzureStorageProducer.php]: Do not throw exception if null is passed

**Comment by makasim** [comment]
add a transport doc just like https://github.com/php-enqueue/enqueue-dev/blob/master/docs/transport/redis.md

**Comment by makasim** [comment]
It needs more tests, at least spec tests should be implemented

**Comment by makasim** [comment]
Code styles should be fixed, you could run php-cs-fixer to auto-fix them.

**Comment by makasim** [comment]
root phpunit.xml.dist file should be updated too, so tests are executed by travis.

**Comment by PaulRouss3l** [comment]
I'm currently implementing the tests, and will update with the comment you made above =)

**Comment by PaulRouss3l** [comment]
still wip

File: pkg/azure-storage/AzureStorageConsumer.php [hunk 4]
```diff
@@ -91,8 +91,7 @@ public function reject(Message $message, bool $requeue = false): void
         InvalidMessageException::assertMessageInstanceOf($message, AzureStorageMessage::class);
 
         if (true === $requeue) {
-            $factory = new AzureStorageConnectionFactory($this->client);
-            $context = $factory->getContext();
+            $context = new AzureStorageContext($this->client);
```
Comment (makasim) [inline pkg/azure-storage/AzureStorageConsumer.php]: I think context should be passed in consumer constructor. The context should have a getClient public method

**Comment by makasim** [comment]
That was decided to release it as a stand-alone package https://github.com/assoconnect/enqueue-azure

Closing PR. Thank you guys for your effort!

**Pull request closed without merging**